src/Domain/Person.php

```diff
@@ -140,49 +175,68 @@ public function getLastName(): string
 
     public function hashPassword(): void
     {
-        $this->password = Password::hash($this->raw_password);
+        if (!empty($this->raw_password)) {
+            $this->password = Password::hash($this->raw_password);
+        }
     }
 
-    public function getPasswordHash(): string
+    public function getPasswordHash(): ?string
     {
         return $this->password;
     }
 
+    public function setStripeCustomerId(?string $stripe_customer_id): void
+    {
+        $this->stripe_customer_id = $stripe_customer_id;
+    }
+
     #[\ReturnTypeWillChange]
     public function jsonSerialize(): array
     {
+        $this->has_password = $this->getPasswordHash() !== null;
+
         $jsonVars = get_object_vars($this);
         $jsonVars['uuid'] = $this->getId()?->toString();
 
         return $jsonVars;
     }
 
     /**
-     * @Assert\Callback()
+     * @Assert\Callback(groups={"new"})
      * @see Person::$captcha_code
-     * @see Person::$raw_password
      */
-    public function validateCaptchaAndRawPasswordSetIfNew(ExecutionContextInterface $context): void
+    public function validateCaptchaIfNew(ExecutionContextInterface $context): void
     {
         // Brand new entity + no captcha solved.
         if (empty($this->id) && empty($this->captcha_code)) {
             $context->buildViolation('Captcha is required to create an account')
                 ->atPath('captcha_code')
                 ->addViolation();
         }
+    }
 
-        // Entity brand new or somehow otherwise without a password, and none set.
-        $passwordMissingOrInvalid = (
-            empty($this->password) &&
-            (empty($this->raw_password) || mb_strlen($this->raw_password) < static::MIN_PASSWORD_LENGTH)
+    /**
+     * @Assert\Callback(groups={"complete"})
+     * @see Person::$raw_password
+     */
+    public function validatePasswordIfNotBlank(ExecutionContextInterface $context): void
+    {
+        $passwordUpdatedAndTooShort = (
```

Comment:
What is someone attempts to update the person with a blank password? Under this validation I presume it'd save fine?

Reply:
@alihejazi I think this is OK. It is allowed to set a `null` password and I think it's simplest to treat all blank inputs the same. Notice that `hashPassword()` skips using the same criterion.